enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -2784,6 +2789,47 @@ func (e *partitionEvictor) sampleGroup() {
 	metrics.PebbleCacheGroupIDSampleCount.With(prometheus.Labels{metrics.GroupID: groupID}).Inc()
 }
 
+func (e *partitionEvictor) randomKeyForEvictionSampling() ([]byte, error) {
+	var sampleType rspb.CacheType
+	e.mu.Lock()
+	switch e.lastTypeSampled {
+	case rspb.CacheType_AC:
+		sampleType = rspb.CacheType_CAS
+	case rspb.CacheType_CAS:
+		sampleType = rspb.CacheType_AC
+	default:
+		sampleType = rspb.CacheType_CAS
+	}
+
+	// Don't allow AC eviction if it's not enabled.
+	if sampleType == rspb.CacheType_AC && !*acEvictionEnabled {
+		sampleType = rspb.CacheType_CAS
+	}
+	e.lastTypeSampled = sampleType
+	e.mu.Unlock()
+
+	// Attempt AC sampling (but do not error out if it fails, instead fall
+	// through to CAS sampling)
+	if sampleType == rspb.CacheType_AC {
+		gid, err := e.randomGroupForEvictionSampling()
+		if err != nil {
+			if !status.IsNotFoundError(err) {
+				log.Warningf("no groups to sample for %q: %s", e.part.ID, err)
+			}
+		} else {
+			if rk, err := e.randomKey(64, gid, sampleType); err == nil {
```

Comment:
Do you need both this and the randomKey call below? If you leave gid as empty if not doing AC sampling you can just have one randomKey call at the end of the function.

Reply:
Done.